ticket-proposal/use_cases.md

```diff
@@ -0,0 +1,13 @@
+# Ticket Events Use Cases
+
+1. Engineering teams usually start with some ticket or issue that allows for
+   features to be implemented or at least begin the engineering process.
+    - This allows management or engineers to start at the ticket and follow
+      life cycle of the actual feature.
+    - Further this allows us to work backwards from an artifact to the ticket  
+2. Allows for metrics regarding tickets, e.g. how many bugs occurred this
+   month, how many were fixed, etc.
+    - How efficiently are we addressing our customers needs? As in when ticket
+      was created to resolved.
+3. Out of scope, but was talked about internally at Apple, is that why not
+   standardize ticket APIs?
```

Comment:
Okay, cool, let's add that. You can also use suggestions if you want to append something within the PR as a non-author so that should be easier to add to the PR as well.

We could probably word it as

```
4. Ticket metadata could be captured from a external system
```

What do you think?

Reply:
I was thinking more about taking an action when a ticket update event is seen.

For more context, take this pipeline as an example:

deploy to dev --> wait for successful tests --> deploy to beta --> wait for successful tests --> wait for approval --> deploy to prod

I need to solve for the "wait for approval" step so we can automatically resume the rollout. The approval comes in as a ticket update.

Reply:
> I need to solve for the "wait for approval" step so we can automatically resume the rollout. The approval comes in as a ticket update.

This is an interesting use case. I wonder what that explicitly means. Like the flow I understand is that approval is not usually done through a ticket. Ticket is usually to highlight a change, bug, etc. So, Im having trouble understanding where approval fits in this picture with tickets. It is perhaps I have a limited understanding of how you guys use tickets, since Im going just based on my experience.

Reply:
Just putting some thoughts down. 

For regulatory purposes we need to capture approvals/sign-offs in Jira. Also, we don't want to deploy an artifact to prod until after it has been approved in beta. 

I can imagine different companies have different approval processes that take place at varying points in the deployment pipeline. I was originally thinking a generic ticket updated event would cover most approval scenarios.